rdmo/views/signals.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from django.contrib.auth.models import User, Group
+from django.contrib.sites.models import Site
+
+from rdmo.questions.models import Catalog
+from rdmo.projects.models import Project, Membership
+
+from .models import View
+
+logger = logging.getLogger(__name__)
+
+def m2m_changed_view_catalog_signal(sender, instance, **kwargs):
+    catalogs = instance.catalogs.all()
+
+    if catalogs.count() > 0:
+        catalog_candidates = Catalog.objects.exclude(id__in=catalogs.values_list('pk')) 
```

Comment:
Here I would use `[catalog.id for catalog in catalogs]`, I am not sure, but I think `values_list` will hit the database again. (nitpicking I know ...)

Reply:
I replaced the `values_list` in favor of corresponding list comprehensions